rocketpy/AeroSurface.py

```diff
@@ -368,11 +313,94 @@ def evaluate_geometrical_parameters(self):
+        p = 3  # Density modifier. Greater n makes more points closer to 0. n=1 -> points equally spaced.
+
+        # Calculate a function to find the tangential intersection point between the circle and nosecone curve.
+        def find_x_intercept(x):
+            return x + self.y_nosecone(x) * self.y_nosecone.differentiate(x)
+
+        # Calculate a function to find the radius of the nosecone curve
+        def find_radius(x):
+            return (self.y_nosecone(x) ** 2 + (x - find_x_intercept(x)) ** 2) ** 0.5
+
+        # Check bluffness circle and choose whether to use it or not
+        if self.bluffness is None or self.bluffness == 0:
+            # Set up parameters to continue without bluffness
+            r_circle, circle_center, x_init = 0, 0, 0
+        else:
+            # Calculate circle radius
+            r_circle = self.bluffness * self.base_radius
+            if self.kind == "elliptical":
+                # Calculate a function to set up a circle at the starting position to test bluffness
+                def test_circle(x):
+                    return np.sqrt(r_circle**2 - (x - r_circle) ** 2)
+
+                # Check if bluffness circle is too small
+                if test_circle(1e-03) <= self.y_nosecone(1e-03):
+                    # Raise a warning
+                    warnings.warn(
+                        "WARNING: The chosen bluffness ratio is too small for "
+                        "the selected nose cone category, thereby the effective "
+                        "bluffness will be 0."
+                    )
+                    # Set up parameters to continue without bluffness
+                    r_circle, circle_center, x_init = 0, 0, 0
+                else:
+                    # Find the intersection point between circle and nosecone curve
+                    x_init = fsolve(lambda x: find_radius(x) - r_circle, r_circle)[0]
+                    circle_center = find_x_intercept(x_init)
+            else:
+                # Find the intersection point between circle and nosecone curve
+                x_init = fsolve(lambda x: find_radius(x) - r_circle, r_circle)[0]
+                circle_center = find_x_intercept(x_init)
+
+        # Calculate a function to create the circle at the correct position
+        def create_circle(x):
+            return abs(r_circle**2 - (x - circle_center) ** 2) ** 0.5
+
+        # Define a function for the final shape of the curve with a circle at the tip
+        def final_shape(x):
+            return self.y_nosecone(x) if x >= x_init else create_circle(x)
+
+        # Vectorize the final_shape function
+        final_shape_vec = np.vectorize(final_shape)
+
+        # Create the vectors X and Y with the points of the curve
+        self.nosecone_x = (self.length - (circle_center - r_circle)) * (
+            np.linspace(0, 1, n) ** p
+        )
+        self.nosecone_y = final_shape_vec(self.nosecone_x + (circle_center - r_circle))
```

Comment:
You name this variable as `nosecone_y` and `y_nosecone` in the same code, which doesn't feel right.
Also, there's no documentation for these attributes.

I'd recommend creating a single atribute called `coords`, this can be a list of: 
- two items: the list of x coords and the list of y coords.
- n items, with each item being a (xi, yi) pair.

If we manage to create this kind of robust system to store the coordinates, the rocket drawing problem will be tackled really quicker.

Reply:
Solved in https://github.com/RocketPy-Team/RocketPy/pull/413/commits/e847406d9df8eb36befc3f1d33fb9e46c25f00e4